### PR TITLE
RFC for WG Charters; Charter Template

### PR DESCRIPTION
**Pull request by thereisnogabe**
Open to any & all feedback!

### CODE REVIEW

**Review by mistermoe — APPROVED** [review]

**Review by leordev — APPROVED** [review]

File: templates/charter.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Scope
+
+What exactly is in scope and out of scope for the working group.
+
+## Deliverables
+
+Enumeration of work products for the group. Be as detailed as possible. 
+
+Make sure to classify the deliverables accordingly (e.g. specification, SDK, research report, etc.). Assign DRIs for each deliverable as necessary. Deliverables should each have clera acceptance criteria" so we know how to measure their completeness.
```
Comment (kirahsapong) [inline templates/charter.md]:
```suggestion
Make sure to classify the deliverables accordingly (e.g. specification, SDK, research report, etc.). Assign DRIs for each deliverable as necessary. Deliverables should each have clear "acceptance criteria" so we know how to measure their completeness.
```

File: rfcs/rfc-0002/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+## Motivation and Background
+Why is this working group necessary? What does it build upon? What does it enable? Is it related to other groups?
+
+## Scope
+What exactly is in scope and out of scope for the working group.
+
+## Deliverables
+Enumeration of work products for the group. Be as detailed as possible. 
+
+Make sure to classify the deliverables accordingly (e.g. specification, SDK, research report, etc.). Assign DRIs for each deliverable as necessary. Deliverables should each have clera acceptance criteria" so we know how to measure their completeness.
```
Comment (kirahsapong) [inline rfcs/rfc-0002/README.md]:
```suggestion
Make sure to classify the deliverables accordingly (e.g. specification, SDK, research report, etc.). Assign DRIs for each deliverable as necessary. Deliverables should each have clear "acceptance criteria" so we know how to measure their completeness.
```

File: rfcs/rfc-0002/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
+# RFC-0002 - Working Group Charters
+
+## Summary
+
+This RFC proposes both some suggestions on how to run Working Groups along with a template for Working Group charters to be used by the TSC to allow for the creation, and maintenance, of well-structured Working Groups. With this proposal, Working Groups are able to be formed with clear scopes-of-work, including work items with reasonable timelines, and accountability by assigning DRIs.
```
Comment (kirahsapong) [inline rfcs/rfc-0002/README.md]:
```suggestion
This Request for Comment (RFC) proposes both some suggestions on how to run Working Groups along with a template for Working Group charters to be used by the Technical Steering Committee (TSC) to allow for the creation, and maintenance, of well-structured Working Groups. With this proposal, Working Groups are able to be formed with clear scopes-of-work, including work items with reasonable timelines, and accountability by assigning DRIs.
```
per [wcag](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-located.html)

File: rfcs/rfc-0002/README.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+In addition to the charter template below, we should adhere to the following processes in managing Working Groups and their charters:
+
+* A Working Group’s charter must reach consensus in the TSC before being considered approved. 
+
+* A Working Group shall be chartered for a limited set of time (e.g. a few months, a year) with clear deliverables. This enables a realistic scope of work and aids in our goal of having a public timeline.
+
+* Charters should be as explicit as possible in describing deliverables and their acceptance criteria to promote a shared understanding of determining when a work item is to be considered complete.
+
+* Working Groups shall share updates through the TSC via the Working Group DRI. Working Groups shall adhere to the TSC's recommendations on regular reviews (whether monthly, quarterly, or on another timeline).
```
Comment (kirahsapong) [inline rfcs/rfc-0002/README.md]:
```suggestion
* Working Groups shall share updates through the TSC via the Working Group Directly Responsible Individual (DRI). Working Groups shall adhere to the TSC's recommendations on regular reviews (whether monthly, quarterly, or on another timeline).
```

File: rfcs/rfc-0002/README.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+**Ad Hoc Teams**: Instead of formal Working Groups, form ad hoc teams for specific tasks or projects. These teams can be more flexible and nimble, adapting quickly to changes without the need for formal charters.
+
+**Task Forces**: Establish task forces for urgent or high-priority initiatives. Task forces can have a narrow focus and short lifespan, providing a more agile approach to addressing specific issues.
+
+**Centralized Management**: Centralize the management of work items under the TSC. This could streamline processes and reduce the overhead associated with multiple Working Groups.
+
+## Prior Art
+
+The proposed charter template was heavily influenced by the [W3C Charter Templateh](https://w3c.github.io/charter-drafts/charter-template.html) as well as the [DIF Charter Documents](https://github.com/decentralized-identity/org/tree/master/Org%20documents/WG%20documents).
```
Comment (jiyoonie9) [inline rfcs/rfc-0002/README.md]: typo :D 
```suggestion
The proposed charter template was heavily influenced by the [W3C Charter Template](https://w3c.github.io/charter-drafts/charter-template.html) as well as the [DIF Charter Documents](https://github.com/decentralized-identity/org/tree/master/Org%20documents/WG%20documents).
```

File: rfcs/rfc-0002/README.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+**Complexity in Coordination**: Managing multiple Working Groups with their own charters could create complexity in coordination, especially if there are interdependencies between groups. A more centralized approach might simplify coordination and reduce potential conflicts.
+
+Counterpoint: This trades off giving more autonomy and flexibility to those closest to the work.
+
+#### Alternative Approaches:
+
+**Ad Hoc Teams**: Instead of formal Working Groups, form ad hoc teams for specific tasks or projects. These teams can be more flexible and nimble, adapting quickly to changes without the need for formal charters.
+
+**Task Forces**: Establish task forces for urgent or high-priority initiatives. Task forces can have a narrow focus and short lifespan, providing a more agile approach to addressing specific issues.
```
Comment (kirahsapong) [inline rfcs/rfc-0002/README.md:93]: Appreciate the alternative approaches here. I'm not sure the distinction between ad hoc teams and task forces but FWIW it's an interesting consideration to begin with an ad hoc/task force team given the overhead of the Working Group. An ad hoc/task force team feels like the equivalent of an MVP compared with a Working Group - we may arrive to needing working groups ultimately, but task forces feel like less overhead and may be sufficient in achieving aims. Not sure, it's a good consideration.

File: rfcs/rfc-0002/README.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+1. How to operate a WG (tooling & meeting/comms norms).
+
+A GitHub repo could be created per WG; though it could also be viable to track WGs via the TSC repository.
+
+Similarly meetings/comms norms could be handled by each group, with common suggestions for async processes (like using GitHub issues and lableing to track work).
+
+2. Handling participation.
+
+Should we track who is participating in each working group? Or stick to a top level DRI for the Working Group and each of its deliverables? How do we handle changes over time?
```
Comment (jiyoonie9) [inline rfcs/rfc-0002/README.md]: per the meeting on 5/30, we're leaning on sticking a top level DRI for the working group who is the tiebreaker, if consensus cannot be reached

**Review by kirahsapong — APPROVED** [review]
Really great detail! Thanks for putting this together.

File: rfcs/rfc-0002/README.md (hunk 6 above)
Comment (thereisnogabe) [reply rfcs/rfc-0002/README.md:93]: I thought of it like -- an ad-hoc team is spun up to _do the work_ (kind of like a scrum team), where a _task force_ would be more exploratory/short lived for a single (usually smaller in scope) piece of work

it is a bit abstract

**Comment by frankhinek** [comment]
@decentralgabe Feedback from May 30th call -- part of this @jiyoontbd already highlighted above

    - @jiyoontbd Called out that we need to ensure that we tie spec changes to a concrete implementation to prove out what is being proposed. We should consider adding some guidance to the Working Group chartering process ensuring that there's a tight feedback loop from implementation work so that we don't end up with specs that are difficult to interpret/implement.
    - @jiyoontbd We should clarify that the DRI model applies to tie breaker decisions for Working Groups

**Comment by thereisnogabe** [comment]
@frankhinek to the first point I believe this should be left to the WG's charter. Some WG's may have specs as deliverables, others may not. If they do then following @jiyoontbd's guidance makes sense.

I'll add a note on the tie breaker comment.

**Comment by thereisnogabe** [comment]
updated the doc and set the status to 'active' I will merge in 24 hours barring any feedback

**Pull request merged**